src/Microsoft.Extensions.Logging.EventSource/EventSourceLoggerProvider.cs

```diff
@@ -12,226 +13,94 @@ namespace Microsoft.Extensions.Logging.EventSource
             for (var logger = _loggers; logger != null; logger = logger.Next)
             {
-                logger.Level = ParseLevelSpecs(filterSpec, _defaultLevel, logger.CategoryName);
-            }
-
-            if (_factoryID == 0)
-            {
-                // Compute an ID for the Factory.  It is its position in the list (starting at 1, we reserve 0 to mean unstarted).
-                _factoryID = 1;
-                for (var cur = Next; cur != null; cur = cur.Next)
-                {
-                    _factoryID++;
-                }
+                logger.Level = GetLoggerLevel(logger.CategoryName);
             }
         }
 
-        private LogLevel GetDefaultLevel()
+        private LogLevel GetLoggerLevel(string loggerCategoryName)
         {
-            var allMessageKeywords = LoggingEventSource.Keywords.Message | LoggingEventSource.Keywords.FormattedMessage | LoggingEventSource.Keywords.JsonMessage;
-
-            if (_eventSource.IsEnabled(EventLevel.Informational, allMessageKeywords))
+            if (!_handleFilters)
             {
-                if (_eventSource.IsEnabled(EventLevel.Verbose, allMessageKeywords))
-                {
-                    return LogLevel.Debug;
-                }
-                else
-                {
-                    return LogLevel.Information;
-                }
+                return LogLevel.Trace;
             }
-            else
-            {
-                if (_eventSource.IsEnabled(EventLevel.Warning, allMessageKeywords))
-                {
-                    return LogLevel.Warning;
-                }
-                else
-                {
-                    if (_eventSource.IsEnabled(EventLevel.Error, allMessageKeywords))
-                    {
-                        return LogLevel.Error;
-                    }
-                    else
-                    {
-                        return LogLevel.Critical;
-                    }
-                }
-            }
-        }
 
-        /// <summary>
-        /// Given a set of specifications  Pat1:Level1;Pat1;Level2 ... Where
-        /// Pat is a string pattern (a logger Name with a optional trailing wildcard * char)
-        /// and Level is a number 0 (Trace) through 5 (Critical).
-        ///
-        /// The :Level can be omitted (thus Pat1;Pat2 ...) in which case the level is 1 (Debug).
-        ///
-        /// A completely emtry sting act like * (all loggers set to Debug level).
-        ///
-        /// The first speciciation that 'loggers' Name matches is used.
-        /// </summary>
-        private LogLevel ParseLevelSpecs(string filterSpec, LogLevel defaultLevel, string loggerName)
-        {
-            if (filterSpec == null)
-            {
-                return LoggingEventSource.LoggingDisabled;      // Null means disable.
-            }
-            if (filterSpec == string.Empty)
+            var rules = _rules;
```

Comment:
Null check, sometimes `SetFilterSpec` is given null

Reply:
it's handled in ParseFilterSpec